### ISSUE TITLE
no-useless-predicate: detect always true literal comparison

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#533]
TypeScript already detects comparisons that will always be false because the types have no overlap.
However, it doesn't detect the opposite: if a comparison of literal types will always be true.
Related upstream feature request: https://github.com/Microsoft/TypeScript/issues/28569

```ts
declare function get<T>(): T;

'a' === 'a'; // always true
'a' == 'a'; // always true
'a' !== 'a'; // always false

'a' === 'b'; // compiler already errors here
'a' !== 'b'; // compiler already errors here
'0' == 0; // compiler already errors here

get<'a' | 'b'>() === 'a'; // valid
get<'a' | 'b'>() === get<'a' | 'b'>(); // valid
get<string> === get<string>(); // valid

// advanced
'0' == get<'0' | 0>(); // always true
'0' == get<'0' | 1>(); // valid

// should be handled by a different rule
let a: string;
a === a;
```

The same applies to `switch` although that should be rather rare.

When https://github.com/Microsoft/TypeScript/issues/26592 is fixed upstream, there might be the need for an additional check for loose equal with values of a different type, e.g. `'0' == 0`.
